daemon/api.go

```diff
@@ -2125,12 +2125,13 @@ func getUserDetailsFromStore(theStore snapstate.StoreService, email string) (str
 	return v.Username, opts, nil
 }
 
-func createAllKnownSystemUsers(st *state.State, createData *postUserCreateData) Response {
+func createAllKnownSystemUsers(o *overlord.Overlord, createData *postUserCreateData) Response {
```

Comment:
I'm not too fond of passing Overlord here just to call Model() and State() on it (but I might be missing something and haven't looked at remaining PRs yet) as IMHO it creates tight coupling with overlord and makes testing potentially harder, and also makes the intent of the arguments less obvious. How about passing state as before, and having a Modeler (ModelGetter) interface  argument instead (the interface would declare just the Model() method)?

Reply:
it's a fair remark, I kind of wondered the same but then the test change didn't get too much in the way (maybe we have too much setup there :) ),  but I can look at this again

Reply:
Could we pass `*asserts.Model` and state explicitly instead?

Reply:
The good thing about making the intent clear via an interface would be preventing of creating more (unintentional) coupling with overlord in the future, but that's maybe a theoretical concern. In any case, I suppose existing tests wouldn't be impacted (much) by such change, so maybe it's worth doing it.

Reply:
Yes, passing `*asserts.Model` is probably the simplest, I might do that in one of the follow up though at this point.